tokio-postgres/src/replication.rs

```diff
@@ -0,0 +1,184 @@
+//! Utilities for working with the PostgreSQL replication copy both format.
+
+use crate::copy_both::CopyBothDuplex;
+use crate::Error;
+use bytes::{BufMut, Bytes, BytesMut};
+use futures::{ready, SinkExt, Stream};
+use pin_project_lite::pin_project;
+use postgres_protocol::message::backend::{LogicalReplicationMessage, ReplicationMessage};
+use postgres_protocol::PG_EPOCH;
+use postgres_types::PgLsn;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::SystemTime;
+
+const STANDBY_STATUS_UPDATE_TAG: u8 = b'r';
+const HOT_STANDBY_FEEDBACK_TAG: u8 = b'h';
+
+pin_project! {
+    /// A type which deserializes the postgres replication protocol. This type can be used with
+    /// both physical and logical replication to get access to the byte content of each replication
+    /// message.
+    ///
+    /// The replication *must* be explicitly completed via the `finish` method.
+    pub struct ReplicationStream {
```

Comment:
I'm a little confused by the last line of the description.  What `finish` method?